content/blog/reds/2020-01-14_haiku_monthly_activity_report_-_012019.md

```diff
@@ -0,0 +1,37 @@
+
+On the coding side, from around hrev53618 to hrev53714 Haiku had seen a lot of fixes and improvements, biggest of which are as always listed into arbitrary categories below...
+
+<h3>Terminal</h3>
+The Terminal app saw a lot of fixes during December. Jérôme Duval fixed the encoding problems with TermView scripting and Zoltán Szatmáry went on a coding spree - he fixed crashes caused by double freed memory in case of changing active keymaps, a few window resize problems and a few other minor issues.
+
+<h3>Interface</h3>
+The UI libraries saw a few fixes - X547 fixed focusing on windows with a special flag, flickering while drawing the background caused by wrong window update order and transparency errors in the app_server.
+
+Kacper Kasper added new optimized bitmap drawing routines, which will be used in webkit for faster rendering of lines and tiled images.
+
+<h3>Game Kit</h3>
+As a GCI task, R4H33M had rewritten the PushGameSound and WindowScreen headers as they were copypasted from old BeOS code.
+
+<h3>Filesystem</h3>
+Pulkomandy merged some code contributed by ahenriksson durign GSoC, which will one day allow for resizing BFS partitions.
+
+<h3>Various</h3>
+X547 fixed Haiku3D crashing on exit by moving the cleanup code to another class.
+
+In an attempt to make Haiku's C library more modern, Waddlesplash rewrote a part of math.h to use GCC built-in functions.
+
+mmlr fixed FFmpeg's leaky input buffer in the case that the file failed to load. This could've lead to leaking at least 32K of memory and potentially much more, depending on how much FFmpeg has already allocated for that file. In a production environment, this could easily lead to memory exhaustion, which could lead to further system instability.
+
+<h3>Are we in beta2 yet?</h3>
+Not quite, but we're getting closer; A quick look on the <a href="https://dev.haiku-os.org/milestone/R1/beta2">bugtracker</a> reveals that there are at least 32 blocking issues left, and most of those are driver bugs. Probably the most nutorious one is the intel_extreme driver, which has problems on several machines with graphic chipsets ranging from the old GMA 9XX once popular in laptops and netbooks up to HD4400 from the Ivy Bridge generation of Intel CPUs. Other than that, there's a bug in XHCI preventing USB storage devices from being detected at boot time, which means that some motherboards are unable to boot Haiku from USB 3.0 onwards. Besides driver issues, net80211's code needs to be tidied up for better future compatibility with wpa_supplicant ported from FreeBSD. Lastly, all files that contain permission syscalls need to be audited, and new security checks need to be implemented.
```

Comment:
nutorious -> notorious

Maybe add a few line-breaks.

Reply:
The intel graphics issues are not blocking. If they are not fixed in beta2, they will be in beta3 or later.

Reply:
They were listed under the beta2 milestone in the bugtracker. If they're not blocking, I kinda don't understand which ones are..

Reply:
There is a "priority" column for which the higher level is "blocker".
We consider "blocker" and "high" priorities to be really important for a release, the other things can be postponed if we change our mind.